File: Mobile-SDK-Android-V5/SampleCode-V5/android-sdk-v5-sample/src/main/java/dji/barcode/BarcodeScanActivity.java
package dji.barcode;

import android.app.Activity;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.widget.Toast;
import java.io.File;
import java.util.List;

public class BarcodeScanActivity extends Activity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // No UI, solo procesamiento automático
        Intent intent = getIntent();
        String imagePath = intent.getStringExtra("image_path");
        boolean scanSuccess = false;
        if (imagePath != null) {
            File imgFile = new File(imagePath);
            if (imgFile.exists()) {
                Bitmap bitmap = BitmapFactory.decodeFile(imagePath);
                BarcodeProcessor processor = new BarcodeProcessor();
                List<String> results = processor.processImage(bitmap);
                if (results != null && !results.isEmpty()) {
                    scanSuccess = true;
                    Toast.makeText(this, "Escaneo exitoso", Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(this, "Escaneo fallido", Toast.LENGTH_SHORT).show();
                }
            }
        }
        Intent result = new Intent();
        result.putExtra("scan_success", scanSuccess);
        result.putExtra("image_path", imagePath);
        setResult(Activity.RESULT_OK, result);
        finish();
    }
}

File: Mobile-SDK-Android-V5/SampleCode-V5/android-sdk-v5-sample/src/main/java/dji/barcode/BarcodeScanActivity.kt
package dji.barcode

import android.app.Activity
import android.content.Intent
import android.graphics.BitmapFactory
import android.os.Bundle
import android.widget.Toast
import java.io.File

class BarcodeScanActivity : Activity() {
    
    companion object {
        const val EXTRA_SCAN_RESULTS = "scan_results"
        const val EXTRA_SCAN_SUCCESS = "scan_success"
        const val EXTRA_IMAGE_PATH = "image_path"
    }
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        
        val imagePath = intent.getStringExtra(EXTRA_IMAGE_PATH)
        var scanSuccess = false
        var scanResults: List<String> = emptyList()
        
        if (imagePath != null) {
            val imgFile = File(imagePath)
            if (imgFile.exists()) {
                try {
                    val bitmap = BitmapFactory.decodeFile(imagePath)
                    if (bitmap != null) {
                        val processor = BarcodeProcessor()
                        scanResults = processor.processImage(bitmap)
                        
                        if (scanResults.isNotEmpty()) {
                            scanSuccess = true
                            val codesFound = scanResults.joinToString(", ")
                            Toast.makeText(this, "Códigos encontrados: $codesFound", Toast.LENGTH_LONG).show()
                        } else {
                            Toast.makeText(this, "No se encontraron códigos en la imagen", Toast.LENGTH_SHORT).show()
                        }
                        bitmap.recycle()
                    } else {
                        Toast.makeText(this, "Error al cargar la imagen", Toast.LENGTH_SHORT).show()
                    }
                } catch (e: Exception) {
                    Toast.makeText(this, "Error al procesar imagen: ${e.message}", Toast.LENGTH_SHORT).show()
                }
            } else {
                Toast.makeText(this, "Archivo de imagen no encontrado", Toast.LENGTH_SHORT).show()
            }
        } else {
            Toast.makeText(this, "Ruta de imagen no especificada", Toast.LENGTH_SHORT).show()
        }
        
        // Devolver resultado
        val result = Intent().apply {
            putExtra(EXTRA_SCAN_SUCCESS, scanSuccess)
            putExtra(EXTRA_IMAGE_PATH, imagePath)
            putStringArrayListExtra(EXTRA_SCAN_RESULTS, ArrayList(scanResults))
        }
        setResult(Activity.RESULT_OK, result)
        finish()
    }
}
